source/s3_client.c

```diff
@@ -393,9 +398,22 @@ struct aws_s3_client *aws_s3_client_new(
     return client;
 
 on_error:
-
-    aws_s3_client_release(client);
-
+    aws_event_loop_group_release(client->body_streaming_elg);
+    client->body_streaming_elg = NULL;
+    if (client->tls_connection_options) {
+        aws_tls_connection_options_clean_up(client->tls_connection_options);
+        aws_mem_release(client->allocator, client->tls_connection_options);
+        client->tls_connection_options = NULL;
+    }
+elg_create_fail:
```

Comment:
I think this should be moved after the event_loop_group_release?

Reply:
Do, you mean after the `aws_event_loop_group_release(client->client_bootstrap->event_loop_group);`? But, this `elg_create_fail` tag will handle the error when body_streaming_elg failed to create, and the `client_config->client_bootstrap->event_loop_group` is acquired before the creation of body_streaming_elg.